src/sqlfluff/dialects/dialect_tsql.py

```diff
@@ -1520,7 +1520,10 @@ class SetStatementSegment(BaseSegment):
             "CONCAT_NULL_YIELDS_NULL",
             "CURSOR_CLOSE_ON_COMMIT",
             "FIPS_FLAGGER",
-            "IDENTITY_INSERT",
+            Sequence(
+                "IDENTITY_INSERT",
+                Ref("TableReferenceSegment")
+            ),
```

Comment:
Need to run Black on your code, but suspect this is the change it'll suggest:

```suggestion
            Sequence("IDENTITY_INSERT", Ref("TableReferenceSegment")),
```